src/Microsoft.Azure.SignalR.Common/RestClients/RestClientBuilder.cs

```diff
@@ -0,0 +1,47 @@
+
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using Microsoft.Rest;
+
+namespace Microsoft.Azure.SignalR
+{
+    internal class RestClientBuilder
+    {
+        private Uri BaseUri;
+        private ServiceClientCredentials Credentials;
+        private readonly List<DelegatingHandler> Handlers = new List<DelegatingHandler>();
+        private AsrsUserAgentHandler AzUserAgentHandler;
+
+        internal RestClientBuilder WithProductInfo(string productInfo)
```

Comment:
are `productInfo` and `serviceEndpoint` mandantory when creating the client? if it is, shall we move them to ctor instead of using `With...`?